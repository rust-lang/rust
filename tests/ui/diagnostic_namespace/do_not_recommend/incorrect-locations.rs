//@ check-pass
#![feature(do_not_recommend)]

#[diagnostic::do_not_recommend]
//~^WARN `#[diagnostic::do_not_recommend]` can only be placed
const CONST: () = ();

#[diagnostic::do_not_recommend]
//~^WARN `#[diagnostic::do_not_recommend]` can only be placed
static STATIC: () = ();

#[diagnostic::do_not_recommend]
//~^WARN `#[diagnostic::do_not_recommend]` can only be placed
type Type = ();

#[diagnostic::do_not_recommend]
//~^WARN `#[diagnostic::do_not_recommend]` can only be placed
enum Enum {}

#[diagnostic::do_not_recommend]
//~^WARN `#[diagnostic::do_not_recommend]` can only be placed
extern "C" {}

#[diagnostic::do_not_recommend]
//~^WARN `#[diagnostic::do_not_recommend]` can only be placed
fn fun() {}

#[diagnostic::do_not_recommend]
//~^WARN `#[diagnostic::do_not_recommend]` can only be placed
struct Struct {}

#[diagnostic::do_not_recommend]
//~^WARN `#[diagnostic::do_not_recommend]` can only be placed
trait Trait {}

#[diagnostic::do_not_recommend]
impl Trait for i32 {}

fn main() {}
