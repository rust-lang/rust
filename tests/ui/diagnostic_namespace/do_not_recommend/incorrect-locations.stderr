warning: `#[diagnostic::do_not_recommend]` can only be placed on trait implementations
  --> $DIR/incorrect-locations.rs:4:1
   |
LL | #[diagnostic::do_not_recommend]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unknown_or_malformed_diagnostic_attributes)]` on by default

warning: `#[diagnostic::do_not_recommend]` can only be placed on trait implementations
  --> $DIR/incorrect-locations.rs:8:1
   |
LL | #[diagnostic::do_not_recommend]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: `#[diagnostic::do_not_recommend]` can only be placed on trait implementations
  --> $DIR/incorrect-locations.rs:12:1
   |
LL | #[diagnostic::do_not_recommend]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: `#[diagnostic::do_not_recommend]` can only be placed on trait implementations
  --> $DIR/incorrect-locations.rs:16:1
   |
LL | #[diagnostic::do_not_recommend]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: `#[diagnostic::do_not_recommend]` can only be placed on trait implementations
  --> $DIR/incorrect-locations.rs:20:1
   |
LL | #[diagnostic::do_not_recommend]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: `#[diagnostic::do_not_recommend]` can only be placed on trait implementations
  --> $DIR/incorrect-locations.rs:24:1
   |
LL | #[diagnostic::do_not_recommend]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: `#[diagnostic::do_not_recommend]` can only be placed on trait implementations
  --> $DIR/incorrect-locations.rs:28:1
   |
LL | #[diagnostic::do_not_recommend]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: `#[diagnostic::do_not_recommend]` can only be placed on trait implementations
  --> $DIR/incorrect-locations.rs:32:1
   |
LL | #[diagnostic::do_not_recommend]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: 8 warnings emitted

